### DISCUSSION TITLE
Goal/scope of databend

### DISCUSSION THREAD

**Discussion by dclong** [databendlabs/databend#3245] [Q&A] [answered]
May I ask the goal/scope of databend. Does it plan to provide SQL-like querying language only or does it plan to provide DataFrame-like API too (similar to what Spark does)? 

And will databend support HDFS in future?

**Comment by bohutang** [comment, marked as answer]
Hi @dclong ,
1. The goal of databend is to provide SQL-like querying language.
2. For supporting HDFS, no plan yet. If anyone interested in it, HDFS is a storage of https://github.com/datafuselabs/databend/tree/8c3ef25e153855ebbf9adbb2380ef6ef034ef902/query/src/storages

**Reply by Xuanwo** [reply]
I'm interested in supporting various storage backend for databend. For example: gcs, hdfs or ipfs.

**Reply by ZhiHanZ** [reply]
GCS storage backend would be an interesting good first issue
